likecoin/js/admin-settings/src/pages/PublishSettingPage.js

```diff
@@ -0,0 +1,285 @@
+              } else {
+                errorMessage = errorMessage.concat(e.message);
+              }
+            });
+          }
+          setMattersLoginError(errorMessage);
+        }
+      } else if (error.message) {
+        let errorMessage = 'ERROR:';
+        errorMessage = errorMessage.concat(error.message);
+        setMattersLoginError(errorMessage);
+      }
+    }
+  }
+
+  async function loginHandler(e) {
+    e.preventDefault();
+    const mattersId = mattersIdRef.current.value;
+    const mattersPassword = mattersPasswordRef.current.value;
+    const data = {
+      mattersId,
+      mattersPassword,
+    };
+    // send to Matters API.
+    await loginToMattersAndSaveDataToWordpress(data);
+  }
+  async function handleMattersLogout(e) {
+    setSavedSuccessful(false);
+    e.preventDefault();
+
+    // set local state
+    setSiteMattersId('');
+
+    const siteMattersUser = {
+      mattersId: '',
+      accessToken: '',
+    };
+
+    // change global state & DB
+    postSiteMattersLogin(siteMattersUser);
+    setSavedSuccessful(true);
+  }
+  function handleNoticeDismiss(e) {
+    e.preventDefault();
+    setSavedSuccessful(false);
+  }
+
+  async function confirmHandler(e) {
+    setSavedSuccessful(false);
+    e.preventDefault();
+    const isSiteMattersAutoSaveDraft = siteMattersAutoSaveDraftRef.current.checked;
+    const isSiteMattersAutoPublish = siteMattersAutoPublishRef.current.checked;
+    const isSiteMattersAddFooterLink = siteMattersAddFooterLinkRef.current.checked;
+    const currentISCNBadgeStyleOption = ISCNBadgeStyleOptionRef.current.value;
+
+    const data = {
+      siteMattersAutoSaveDraft: isSiteMattersAutoSaveDraft,
+      siteMattersAutoPublish: isSiteMattersAutoPublish,
+      siteMattersAddFooterLink: isSiteMattersAddFooterLink,
+      ISCNBadgeStyleOption: currentISCNBadgeStyleOption,
+    };
+
+    // save to Wordpress DB.
+    try {
+      // Change global state & DB
+      postSiteMattersOptions(data);
+      setSavedSuccessful(true);
+    } catch (error) {
+      console.error('Error occured when saving to Wordpress DB: ', error); // eslint-disable-line no-console
+      setSavedSuccessful(false);
+    }
+  }
+
+  useEffect(() => {
+    setSiteMattersId(DBSiteMattersId);
+    setSiteMattersAutoSaveDraft(DBSiteMattersAutoSaveDraft);
+    setSiteMattersAutoPublish(DBSiteMattersAutoPublish);
+    setSiteMattersAddFooterLink(DBSiteMattersAddFooterLink);
+    setISCNBadgeStyleOption(DBISCNBadgeStyleOption);
+  }, [
+    DBSiteMattersId,
+    DBSiteMattersAutoSaveDraft,
+    DBSiteMattersAutoPublish,
+    DBSiteMattersAddFooterLink,
+    DBISCNBadgeStyleOption,
+  ]);
+
+  useEffect(() => {
+    setSiteMattersAutoSaveDraft(DBSiteMattersAutoSaveDraft);
+    setSiteMattersAutoPublish(DBSiteMattersAutoPublish);
+    setSiteMattersAddFooterLink(DBSiteMattersAddFooterLink);
+  }, [
+    DBSiteMattersAutoSaveDraft,
+    DBSiteMattersAutoPublish,
+    DBSiteMattersAddFooterLink,
+  ]);
```

Comment:
Can be merged into one?

Reply:
Yes, makes sense! Totally overlooked. :P